src/libraries/System.Private.CoreLib/src/System/Char.cs

```diff
@@ -1000,7 +993,7 @@ public static bool IsSurrogatePair(string s, int index)
                 ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.index);
             }
 
-            if (index + 1 < s.Length)
+            if ((uint)(index + 1) < (uint)s.Length)
```

Comment:
Why is this cast needed, the above check should have ensured `index` is non-negative and the length of a string can't be `int.MaxValue`?

Reply:
It may not be with recent improvements.  It was on .NET 7, according to [SharpLab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgBEBLbAcwDsJcGDmFwsAwhAAmMAIK9sAGwCeuDrgDcNGsQDMTUgzEMA3jQbmme5kgbAIEBQwCSuAMoBXKFAjdsGGAAK2BxQ5AAUzIy4aAwcvBixvNIIAJRmFqbUFtmxAGYMYbgMALzFDLzuCgppWTnmmXV1ACoAFt4A7mEpmrV1AL7pdRz5YWHucRgpcckMAHxlYxMpogAyMLzcGC01jSaDuwytHV09uwPU+9nDBdMwCAwA1AzkDAA8DKvrm9uXGb91xAA7AwwC1sFAWC4PF4fH5AsEoIUANq3BAAXRiuBRSTuj2eaO6/wY512QIYuUUuBgpwsJIs+10THINjsDmcbk83l8/iCIVIEXIURiE0SyR2f16OWuhRKZQqVXFdQaB3MRwgnUJkuydKGI0W8SmOPu8wK4wNnw2W0VOWVKrVGpp/S0Wos0v1kzCqLx5BSb1NSwt32t2VtBzJoPBkI5MO58JCyNRGI+2JmT3IBMdOR1OTJFIUVMz2f2SLYEBguAAchAMAAlGAYTy8NH7AAOUA4ADc4UybMQUIc2uquiVZgwth1yjB2gwAKIIMAwFtCCC8E40PpAA=).

Reply:
Yeah, not handled even in Main 🙁 looking at it